jgiven-core/src/main/java/com/tngtech/jgiven/impl/StageLifecycleManager.java

```diff
@@ -66,6 +72,33 @@ private void executeLifecycleMethods(LifecycyleMethodManager<?> register, boolea
         }
     }
 
+    @SuppressWarnings("unchecked")
+    private void checkGuaranteedStatesAreInitialized() throws JGivenMissingGuaranteedScenarioStateException {
+        for (Field field: FieldCache.get(instance.getClass())
+                                    .getFieldsWithAnnotation(ProvidedScenarioState.class, ScenarioState.class)) {
+            if (field.isAnnotationPresent(ProvidedScenarioState.class)) {
```

Comment:
these double if blocks can go into a separate method, such that you can reuse them

Reply:
Not sure if we can make that work without either using instance of and hence another two if statements or method execution through resolution. You need to cast to that class (ProvidedScenarioState or ScenarioState) in order to be able to run guaranteed() on the object.

Reply:
Yes, I forgot that Annotations don't have inheritance